### PR TITLE
Ribolands

### PR DESCRIPTION
**Pull request by s-will**
Bioconda requires reviews prior to merging pull-requests (PRs). To facilitate this, once your PR is passing tests and ready to be merged, please add the `please review & merge` label so other members of the bioconda community can have a look at your PR and either make suggestions or merge it. Note that if you are not already a member of the bioconda project (meaning that you can't add this label), please ping `@bioconda/core` so that your PR can be reviewed and merged (please note if you'd like to be added to the bioconda project). Please see https://github.com/bioconda/bioconda-recipes/issues/15332 for more details.

* [X] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [X] This PR adds a new recipe.
* [X] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by druvus** [comment]
@BiocondaBot please update

**Comment by eggzilla** [comment]
Hi @s-will  I will have a look :-)

**Comment by eggzilla** [comment]
Is it possible that the shebang of drtransformer.py is breaking the build?

> 14:07:55 BIOCONDA INFO (OUT) /opt/conda/conda-bld/ribolands_1612965590876/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/bin/DrTransformer.py: line 7: from: command not found

**Comment by eggzilla** [comment]
https://github.com/bad-ants-fleet/ribolands/blob/master/scripts/DrTransformer.py
I think you need a insert a
`#! /usr/bin/env python`

**Comment by s-will** [comment]
HI @eggzilla, I agree, this seems the current (hopefully last remaining) problem. Many thanks for finding this and sorry for inviting to review prematurely (I was too optimistic ;)). I also think that patching the shebang is the best solution.

File: recipes/ribolands/meta.yaml [hunk 1]
```diff
@@ -27,7 +26,6 @@ requirements:
     - matplotlib-base
     - future
     - crnsimulator
-    - treekin ==0.5.1
   run:
     - python <3.9
```
Comment (eggzilla) [inline recipes/ribolands/meta.yaml:30]: Do you really need the upper bound?

File: recipes/ribolands/meta.yaml [hunk 2]
```diff
@@ -36,7 +34,6 @@ requirements:
     - matplotlib-base
     - future
     - crnsimulator
-    - treekin ==0.5.1
 
 test:
```
Comment (eggzilla) [inline recipes/ribolands/meta.yaml:38]: If possible a more extensive test then --version would be awesome :-)

Comment (eggzilla) [reply recipes/ribolands/meta.yaml:38]: @s-will : Please comment concerning the upper version bound on python and optionally adding a test.

**Comment by s-will** [comment]
Hi @eggzilla it passed the test, but I cannot produce a Mac treekin 0.5.1, because mlapack is not available.

**Comment by eggzilla** [comment]
Do you want to temporarily add a skip for MacOs?

**Comment by s-will** [comment]
I skip MacOs now. Maybe I would return later to check again, why mlapack failed.

**Comment by s-will** [comment]
For the ribolands pacakge there is an interesting problem, which I was not sure how to best resolve - the package must be architecture independent, since it is pure python (otherwise lint complains). But this means that I can neither skip osx (lint complains) or define treekin 0.5.1 as dependency (test on osx is going to fail).

**Comment by s-will** [comment]
I resolved it by dropping the treekin dependency in ribolands, which is not ideal, but at least makes everything usable, so I think we can justify it

**Comment by s-will** [comment]
@eggzilla concerning your two remarks to the code: 1) the above also means that the pin to 0.5.1 is dropped. Still doctor transformer, one of the scripts in ribolands requires it, because previous versions didn't have mlapack support. 2) What do you have in mind? (Actually, I thought it's common practice to only run the tools with help or version in conda tests... I can of course call the other scripts in the package as well, or directly test import of the python modules)

**Comment by eggzilla** [comment]
If you like it would be awesome if you could add calls with --version for the other scripts. 
Concerning the noarch & skip situation, I think you can use a lint-skip for no-arch python:
https://bioconda.github.io/contributor/linting.html#skipping-a-lint-check
https://bioconda.github.io/contributor/linting.html#id14
I guess like this:
extra:
  skip-lints:
    - should_be_noarch_python

Then you should be able to use skip with macos again :-)

**Comment by s-will** [comment]
cool. thanks, I can try the skip-lint (later) and add tests

**Review by eggzilla — APPROVED** [review]
Looks good :-)

**Comment by s-will** [comment]
:+1: (I forgot the to explain the strange viennarna PIN; it seems that this is temporarily required, since the viennarna does not provide Python bindings for 3.9 - I tried to fix this by rebuilding viennarna, but this didn't succeed and I cannot see why.)

**Comment by eggzilla** [comment]
Oh, I hope I did not merge prematurely ?!
I guess you are referring to this issue: https://github.com/ViennaRNA/ViennaRNA/issues/110 ?

**Comment by s-will** [comment]
yes, this issue is discussing the same problem with the current viennarna conda package. I don't think the merge is premature - IMO it's more important to finally have the treekin and ribolands conda package. The pin can still be removed in a new ribolands package build when the viennarna issue is resolved.

**Comment by eggzilla** [comment]
ok :-) Ping me if I can help in future updates

**Pull request merged**